### PR TITLE
Auto-generate documents for Script Resolver functions

### LINKED ISSUE #303: Auto-generate docs for functions added to Cottle
These are added in `EDDI\SpeechResponder\ScriptResolver.cs ScriptResolver.buildStore()`

Use similar tech as with the Events to emit wiki documentation from test fixtures.

Index page already manually created at https://github.com/EDCD/EDDI/wiki/Functions, ready for replacement by an auto-generated page.

### PR DESCRIPTION
**Pull request by Tkael**
Resolves #303
- Re-organizes custom script resolver functions into an interface and class structure
- Moves the script resolver itself into a sub-folder
- Generates Help.md and Functions.md automatically from classes during the post-build sequence

### CODE REVIEW

File: SpeechResponder/ScriptResolverService/ICustomFunction.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            if (parameter1 is ScriptResolver resolverParameter)
+            {
+                resolver = resolverParameter;
+            }
+            if (parameter2 is BuiltinStore storeParameter)
+            {
+                store = storeParameter;
+            }
+        }
+    }
```
Comment (Tkael) [inline SpeechResponder/ScriptResolverService/ICustomFunction.cs:42]: We're using the `ResolverInstance` abstract class to implement nesting / recursion for functions which require it.

File: postBuildTests.bat [hunk 2]
```diff
@@ -17,7 +17,7 @@ IF %buildConfiguration%=="Release" (
   SET "testCaseFilter=^/TestCaseFilter:""TestCategory!=Speech"""
 ) ELSE (
   :: Run just our Credentials test
```
Comment (richardbuckle) [inline postBuildTests.bat]: Need to update the comment, or simply remove it as it's not really adding much.

**Review by richardbuckle — CHANGES_REQUESTED** [review]
I haven't read every single wrapped functionin detail.

**Comment by richardbuckle** [comment]
Apart from the stale comment I'm happy.

File: SpeechResponder/CustomFunctions/BlueprintDetails.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        public string description => @"
+This function will provide full information for a blueprint, given its name and grade.
+
+BlueprintDetails() takes two mandatory arguments: the name of the blueprint and the grade to retrieve. 
+
+Common usage of this is to provide further information about a blueprint, for example:
+
+    {set blueprint to BlueprintDetails(""Dirty Drive Tuning"", 5)}
+    {len(blueprint.materials)} {if len(blueprint.materials) > 1: different materials are |else: material is} required to produce {blueprint.localizedName}.";
+        public NativeFunction function => new NativeFunction((values) =>
```
Comment (richardbuckle) [inline SpeechResponder/CustomFunctions/BlueprintDetails.cs:16]: Sorry, a thought just struck me. Isn't this allocating a new instance of the NativeFunction every time the property is accessed? Don't we instead want to say:

```cs
    public NativeFunction function = new NativeFunction((values) =>
```
Or is this some form of [lazy initialization](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization)?

**Pull request merged**